libimage/search.go

```diff
@@ -348,6 +331,41 @@ func searchRepositoryTags(ctx context.Context, sys *types.SystemContext, registr
 	return paramsArr, nil
 }
 
+// getDockerAuthConfig extracts a docker auth config from the CopyOptions.  Returns
+// nil if no credentials are set.
+func (options *SearchOptions) getDockerAuthConfig() (*types.DockerAuthConfig, error) {
+	authConf := &types.DockerAuthConfig{IdentityToken: options.IdentityToken}
+
+	if options.Username != "" {
+		if options.Credentials != "" {
+			return nil, errors.New("username/password cannot be used with credentials")
+		}
+		authConf.Username = options.Username
+		authConf.Password = options.Password
+		return authConf, nil
+	}
+
+	if options.Credentials != "" {
+		split := strings.SplitN(options.Credentials, ":", 2)
+		switch len(split) {
+		case 1:
+			authConf.Username = split[0]
+		default:
+			authConf.Username = split[0]
+			authConf.Password = split[1]
+		}
+		return authConf, nil
+	}
+
+	// We should return nil unless a token was set.  That's especially
+	// useful for Podman's remote API.
+	if options.IdentityToken != "" {
+		return authConf, nil
```

Comment:
It would be more symmetric to only set `authConf.IdentityToken` here, and not at the very top of the function.

(Right now, I would even recommend `return &types.DockerAuthConfig{…}` in the three cases, without having a function-scope `authConf` at all. But I suppose that would not be applicable with the suggested ambiguity detection.)